### PR TITLE
fix: Incorrect JoystickComponent position in landscape mode #2387

### PR DESCRIPTION
**Pull request by cigjonser**
# Description

When in landscape mode, a Component will need to be repositioned based on the new canvas
dimensions. If a Component has not reached the mounted state when
HudMarginComponent.onGameResize is called, then _updateMargins will not be called, and
the position will not be updated.
This PR removes the check that the Component has reached the mounted state in
HudMarginComponent.onGameResize so that _updateMargins will always be called.

## Checklist

- [x] I have followed the [Contributor Guide] when preparing my PR.
- [-] I have updated/added tests for ALL new/updated/fixed functionality.
- [-] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [-] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [x] No, this PR is not a breaking change.

## Related Issues

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org
[CHANGELOG]: https://github.com/flame-engine/flame/blob/main/CHANGELOG.md

### CODE REVIEW

**Review by spydon — APPROVED** [review]

**Review by luanpotter — APPROVED** [review]
LGTM

**Review by erickzanardo — APPROVED** [review]
Would be nice to have a test for this bug. So it doesn't happen again

**Review by erickzanardo — APPROVED** [review]

**Comment by spydon** [comment]
Thanks for your contribution @cigjonser! :)

File: packages/flame/test/components/joystick_component_test.dart [hunk 1]
```diff
@@ -37,6 +37,20 @@ void main() {
...
+      game.onGameResize(Vector2(100, 200));
+      final joystick = JoystickComponent(
+        knob: CircleComponent(radius: 5.0),
+        size: 20,
+        margin: const EdgeInsets.only(left: 20, bottom: 20),
+      );
+      joystick.loaded.then((_) => game.onGameResize(Vector2(200, 100)));
+      await game.ensureAdd(joystick);
+      expect(joystick.position, Vector2(30, 70));
+    });
```
Comment (luanpotter) [inline packages/flame/test/components/joystick_component_test.dart:53]: trailing comma? :P

**Review by luanpotter — APPROVED** [review]
LGTM

File: packages/flame/test/components/joystick_component_test.dart (hunk 1 above)
Comment (spydon) [reply packages/flame/test/components/joystick_component_test.dart]: Very true, fixed 😄

**Pull request merged**